azkaban-common/src/main/java/azkaban/project/JdbcProjectLoader.java

```diff
@@ -116,6 +129,38 @@ public JdbcProjectLoader(Props props) {
     return projects;
   }
 
+  private List<Project> fetchAllInactiveProjects(Connection connection)
+          throws ProjectManagerException {
+    QueryRunner runner = new QueryRunner();
+
+    ProjectResultHandler handler = new ProjectResultHandler();
+    List<Project> projects = null;
+    try {
+      projects =
+              runner.query(connection,
+                      ProjectResultHandler.SELECT_ALL_INACTIVE_PROJECTS, handler);
```

Comment:
Problem: We are loading everything since inception
Possible solution: We can set high limit on number of older entries we are purging (make this configurable). Estimate it as say (60days x Peak_Num_Projects_per_day)
See if there are better ways to solve